src/pages/HomePage.jsx

```diff
@@ -0,0 +1,171 @@
+import React, { useState, useEffect } from "react";
+import { Link, useHistory } from "react-router-dom";
+
+import DeleteModal from "./DeleteModal";
+import TaskDetailModal from "./TaskDetailModal";
+
+import "../styles/HomePage.css";
+
+const HomePage = () => {
+  const [tasks, setTasks] = useState([]);
+  const [isDeleteModalOpen, setIsDeleteModalOpen] = useState(false);
+  const [deletingTaskId, setDeletingTaskId] = useState(null);
+  const [filterBy, setFilterBy] = useState("priority");
+  const [selectedTask, setSelectedTask] = useState(null);
+  const history = useHistory();
+
+  useEffect(() => {
+    const savedTasks = JSON.parse(localStorage.getItem("tasks")) || [];
+    setTasks(savedTasks);
+  }, []);
+
+  const handleDeleteTask = (taskId) => {
+    setIsDeleteModalOpen(true);
+    setDeletingTaskId(taskId);
+  };
+
+  const handleEditTask = (taskId) => {
+    history.push(`/edit-task/${taskId}`);
+  };
+
+  const confirmDeleteTask = () => {
+    const updatedTasks = tasks.filter((task) => task.id !== deletingTaskId);
+    setTasks(updatedTasks);
+    localStorage.setItem("tasks", JSON.stringify(updatedTasks));
+    setIsDeleteModalOpen(false);
+  };
+
+  const cancelDeleteTask = () => {
+    setIsDeleteModalOpen(false);
+    setDeletingTaskId(null);
+  };
+
+  const toggleTaskCompletion = (taskId) => {
+    const updatedTasks = tasks.map((task) =>
+      task.id === taskId ? { ...task, completed: !task.completed } : task
+    );
+    setTasks(updatedTasks);
+    localStorage.setItem("tasks", JSON.stringify(updatedTasks));
+  };
+
+  const daysFromToday = (dueDate) => {
+    const oneDay = 24 * 60 * 60 * 1000; // hours*minutes*seconds*milliseconds
+    const today = new Date();
+    const due = new Date(dueDate);
+    const diffDays = Math.round((due - today) / oneDay);
+    return diffDays;
+  };
+
+  const daysToDisplay = (dueDate) => {
+    const diffDays = daysFromToday(dueDate);
+    if (diffDays < 0) {
+      return `${Math.abs(diffDays)} days ago`;
+    } else {
+      return `${diffDays} days left`;
+    }
+  };
```

Comment:
You can segregate these helper functions. You can create a separate folder called helper and then you can place these function in the date.helper.ts file